### PR TITLE
Fix: Cross-namespace RequestAuthentication

### LINKED ISSUE #54051: Cross-namespace Waypoint Missing RequestAuthn Config
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

With RequestAuthn (bound to svc) and the Service (and also AuthzPol bound to the svc) in the same namespace and the waypoint's gateway resource in a second namespace it appears the RA config is missing in envoy. I see the authz policy in the envoy config but there is no inlined jwks.

### Version

```Text
CLI version:
version.BuildInfo{Version:"1.24.0", GitRevision:"8825a6b7f8c9a2d66005a5f8b64e98aaee0dda99", GolangVersion:"go1.23.2", BuildStatus:"Clean", GitTag:"1.24.0"}

The following Istio control plane revisions/versions were found in the cluster:
Revision 1-24-0:
&version.MeshInfo{version.ServerInfo{Component:"pilot", Revision:"1-24-0", Info:version.BuildInfo{Version:"1.24.0", GitRevision:"8825a6b7f8c9a2d66005a5f8b64e98aaee0dda99", GolangVersion:"", BuildStatus:"Clean", GitTag:"1.24.0"}}}

The following proxy revisions/versions were found in the cluster:
Revision 1-24-0: Versions {1.24.0}
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ilrudie**
**Please provide a description of this PR:**

fixes #54051 

adds searching svc namespace to enable cross-namespace reqauthn

### CODE REVIEW

**Comment by ilrudie** [comment]
Should we consider a 1.24 cherry-pick? I think its fair to expect this would work in 1.24

**Review by howardjohn — COMMENTED** [review]
Given the number of bugs around this and similar it may be good to add a higher level test like `TestWaypointTelemetry`. But this looks good

cc @keithmattix

**Comment by ilrudie** [comment]
cool cool, if we like the general direction I'll take a look at getting more tests in place, ty

**Comment by ilrudie** [comment]
Got some `TestWaypointTelemetry` style tests for RA and x-namespace RA

File: pilot/pkg/xds/waypoint_test.go [hunk 1]
```diff
@@ -306,6 +306,145 @@ spec:
...
+
+	d, proxy := setupWaypointTest(t,
+		waypointGateway, waypointSvc, waypointInstance,
+		appServiceEntry, requestAuthn)
+
+	l := xdstest.ExtractListener("main_internal", d.Listeners(proxy))
+	app := xdstest.ExtractHTTPConnectionManager(t,
+		xdstest.ExtractFilterChain(model.BuildSubsetKey(model.TrafficDirectionInboundVIP, "http", "app.com", 80), l))
+
+	assert.Equal(t, sets.New(slices.Map(app.HttpFilters, (*hcm.HttpFilter).GetName)...).Contains("envoy.filters.http.jwt_authn"), true)
```
Comment (howardjohn) [inline pilot/pkg/xds/waypoint_test.go:445]: nit: would be nice to also test a different service doesn't have it. I think the Telemetry test does this. This ensures we are not accidentally ignoring the Service constraint

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
Hold incase you want to change the test, but I am good as is as well

**Comment by ilrudie** [comment]
> Hold incase you want to change the test, but I am good as is as well

Thanks, I like the suggestion. I'll take a look at getting that added.

File: pilot/pkg/model/authentication.go [hunk 2]
```diff
@@ -238,6 +240,9 @@ func getConfigsForWorkload(rootNamespace string, configsByNamespace map[string][
 	} else {
 		lookupInNamespaces = []string{namespace}
 	}
+	if serviceNamespace != "" && !slices.Contains(lookupInNamespaces, serviceNamespace) {
+		lookupInNamespaces = append(lookupInNamespaces, serviceNamespace)
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/authentication.go:244]: RequestAuthentication is in same ns with workload. Not to mention the selector field, so i believe this is not expected to work well

Comment (ilrudie) [reply pilot/pkg/model/authentication.go:244]: This should be a no-op if the RA is in the same ns as the workload/waypoint but in the cross-namespace waypoint case the RA would be in the same ns as the service/se it's targeting and the waypoint would be in a different namespace. This is the setup which is meant to be covered by this code.

Comment (hzxuzhonghu) [reply pilot/pkg/model/authentication.go:244]: Yeah, i know this case. But if a ra is set with selector, how do we match?

Comment (ilrudie) [reply pilot/pkg/model/authentication.go:244]: We shouldn't match. Selector shouldn't bind to waypoint.

Comment (hzxuzhonghu) [reply pilot/pkg/model/authentication.go:244]: OK, so it means we cannot specify selector in ambient, only targetRef is supported, right?

Comment (ilrudie) [reply pilot/pkg/model/authentication.go:244]: yup, waypoints only support gtwapi-style policy attachment.

**Pull request merged**